ros/lib/kessel.py

```diff
@@ -0,0 +1,171 @@
+from typing import Optional
+from dataclasses import dataclass
+import grpc
+from kessel.inventory.v1beta2 import (
+    inventory_service_pb2_grpc,
+    streamed_list_objects_request_pb2,
+    representation_type_pb2,
+    subject_reference_pb2,
+    resource_reference_pb2,
+    reporter_reference_pb2,
+    request_pagination_pb2,
+    check_request_pb2,
+    allowed_pb2
+)
+
+from enum import Enum
+
+class Allowed(Enum):
+    UNSPECIFIED = 0
+    TRUE = 1
+    FALSE = 2
+
+
+@dataclass
+class ObjectType:
+    resource_type: str
+    reporter_type: str
+
+    @staticmethod
+    def workspace():
+        return ObjectType(
+            resource_type="workspace",
+            reporter_type="rbac",
+        )
+
+    @staticmethod
+    def role():
+        return ObjectType(
+            resource_type="role",
+            reporter_type="rbac"
+        )
+
+
+@dataclass
+class Resource(ObjectType):
+    resource_id: str
+
+    @staticmethod
+    def principal(resource_id: str):
+        return Resource(
+            resource_id=resource_id,
```

Comment:
One correction here, it appears that our resource ids are going to be prefixed with `redhat` so this should be:

```suggestion
            resource_id=f"redhat/{resource_id}",
```